libdnf/sack/query.cpp

```diff
@@ -769,6 +879,64 @@ Query::addFilter(HyNevra nevra, bool icase)
     return 0;
 }
 
+void
+Query::Impl::filterNevraStrict(int cmpType, const char **matches)
+{
+    Pool *pool = dnf_sack_get_pool(sack);
+    std::vector<NevraID> compareSet;
+    const unsigned nmatches = g_strv_length((gchar**)matches);
+    compareSet.reserve(nmatches);
+    NevraID nevraId;
+    for (unsigned int i = 0; i < nmatches; ++i) {
+        const char * nevraPattern = matches[i];
+        if (!nevraPattern)
+            throw std::runtime_error("Query can not accept NULL for STR match");
+        if (nevraId.parse(pool, nevraPattern)) {
+            compareSet.push_back(nevraId);
+        }
+    }
+    Map nevraResult;
```

Comment:
If the compareSet vector is empty, there is no need to allocate nevraResult bitmap.
Probably we can insert this code before the line:
```
    if (compareSet.empty()) {
        if (!(cmpType & HY_NOT))
            map_empty(result->getMap());
        return;
    }
```